ceph_iscsi_config/gateway.py

```diff
@@ -408,7 +410,9 @@ def map_luns(self, config):
             for tpg in self.tpg_list:
                 self.logger.debug("processing tpg{}".format(tpg.tag))
 
-                if not self.lun_mapped(tpg, stg_object):
+                target_config = config.config["targets"][self.iqn]
+                if not self.lun_mapped(tpg, stg_object) \
+                        and stg_object.name in target_config['disks']:
```

Comment:
I think you want to change the top loop. Instead of looping over every SO just start disks defined for the specific target.

Reply:
Fixed